src/VisualStudio/Core/Def/Implementation/LanguageClient/AbstractInProcLanguageClient.cs

```diff
@@ -176,5 +186,78 @@ public Task OnServerInitializeFailedAsync(Exception e)
             // We don't need to provide additional exception handling here, liveshare already handles failure cases for this server.
             return Task.CompletedTask;
         }
+
+        internal static async Task<InProcLanguageServer> CreateAsync(
+            AbstractInProcLanguageClient languageClient,
+            Stream inputStream,
+            Stream outputStream,
+            ILspWorkspaceRegistrationService lspWorkspaceRegistrationService,
+            VSShell.IAsyncServiceProvider? asyncServiceProvider,
+            string? clientName,
+            CancellationToken cancellationToken)
+        {
+            var jsonMessageFormatter = new JsonMessageFormatter();
+            VSExtensionUtilities.AddVSExtensionConverters(jsonMessageFormatter.JsonSerializer);
+
+            var jsonRpc = new JsonRpc(new HeaderDelimitedMessageHandler(outputStream, inputStream, jsonMessageFormatter));
+            var serverTypeName = languageClient.GetType().Name;
+            var logger = await CreateLoggerAsync(asyncServiceProvider, serverTypeName, clientName, jsonRpc, cancellationToken).ConfigureAwait(false);
+
+            var server = languageClient.Create(
+                jsonRpc,
+                languageClient.GetCapabilities(),
+                lspWorkspaceRegistrationService,
+                logger ?? NoOpLspLogger.Instance);
+
+            jsonRpc.StartListening();
+            return server;
+        }
+
+        private static async Task<LogHubLspLogger?> CreateLoggerAsync(
```

Comment:
as above, entirely copied from InProcLanguageServer to initialize the VS dependent logger in the VS layer.